app/lib/request_utils.rb

```diff
@@ -79,6 +79,18 @@ def self.sanitize_search_terms(terms)
     SANITIZER.sanitize(inputs).encode!(Encoding.find('ASCII-8BIT'), invalid: :replace, undef: :replace)
   end
 
+  # takes a comma-delimited string of ids (e.g. StudyFile ids) and returns an array of ids
+  # raises Argument error if any of the strings are not valid ids
+  def self.validate_id_list(id_list_string)
```

Comment:
This looks like it works fine, but there's an easier way to check IDs.  Since what we're actually trying to validate is "does this string represent a `StudyFile` id", and all instances of such are `BSON::ObjectId` objects, we can simply cast them:
```
ids = id_list_string.split(',').map(&:strip) # we need to strip any whitespace or it can error out
ids.map {|id| BSON::ObjectId.from_string(id) }
ids
```
If any id is not valid, it will raise an `BSON::ObjectId::Invalid` error.

Reply:
nice, agree that's much cleaner

Reply:
Come to think of it, we actually don't even need the `ids` at the end since the output of `map()` is the same thing and Ruby will auto-return that...